okio/src/main/java/okio/RealBufferedSource.java

```diff
@@ -233,6 +233,51 @@ public RealBufferedSource(Source source) {
     return buffer.readLongLe();
   }
 
+  @Override public long readDecimalLong() throws IOException {
+    int pos = 0;
+    while (true) {
+      if (!request(pos + 1)) {
+        break; // No more data.
+      }
+      byte b = buffer.getByte(pos);
+      if ((b < '0' || b > '9') && (pos != 0 || b != '-')) {
+        break; // Non-digit, or non-leading negative sign.
+      }
+      if (++pos > 20) {
+        break; // Exceeded valid digit count.
```

Comment:
I think this might be incorrect for input like `0000000000000000000000000000001` because we might not block enough